src/test/test-seccomp.c

```diff
@@ -805,67 +806,95 @@ static void test_load_syscall_filter_set_raw(void) {
+                } else {
+                        assert_se(seccomp_load_syscall_filter_set_raw(SCMP_ACT_ALLOW, s, SCMP_ACT_ERRNO(EUCLEAN), true) >= 0);
 
-                assert_se(access("/", F_OK) < 0);
-                assert_se(errno == EUCLEAN);
+                        assert_se(access("/", F_OK) < 0);
+                        assert_se(errno == EUCLEAN);
 
-                assert_se(poll(NULL, 0, 0) == 0);
+                        assert_se(poll(NULL, 0, 0) == 0);
 
-                s = hashmap_free(s);
+                        hashmap_clear(s);
 
-                assert_se(s = hashmap_new(NULL));
 #if defined __NR_access && __NR_access >= 0
-                assert_se(hashmap_put(s, UINT32_TO_PTR(__NR_access + 1), INT_TO_PTR(EILSEQ)) >= 0);
-#else
-                assert_se(hashmap_put(s, UINT32_TO_PTR(__NR_faccessat + 1), INT_TO_PTR(EILSEQ)) >= 0);
+                        assert_se(hashmap_put(s, UINT32_TO_PTR(__NR_access + 1), INT_TO_PTR(EILSEQ)) >= 0);
+#endif
+#if defined __NR_faccessat && __NR_faccessat >= 0
+                        assert_se(hashmap_put(s, UINT32_TO_PTR(__NR_faccessat + 1), INT_TO_PTR(EILSEQ)) >= 0);
 #endif
 
-                assert_se(seccomp_load_syscall_filter_set_raw(SCMP_ACT_ALLOW, s, SCMP_ACT_ERRNO(EUCLEAN), true) >= 0);
+                        assert_se(seccomp_load_syscall_filter_set_raw(SCMP_ACT_ALLOW, s, SCMP_ACT_ERRNO(EUCLEAN), true) >= 0);
 
-                assert_se(access("/", F_OK) < 0);
-                assert_se(errno == EILSEQ);
+                        assert_se(access("/", F_OK) < 0);
+                        assert_se(errno == EILSEQ);
 
-                assert_se(poll(NULL, 0, 0) == 0);
+                        assert_se(poll(NULL, 0, 0) == 0);
 
-                s = hashmap_free(s);
+                        hashmap_clear(s);
+                }
 
-                assert_se(s = hashmap_new(NULL));
+                assert_se(hashmap_isempty(s));
 #if defined __NR_poll && __NR_poll >= 0
                 assert_se(hashmap_put(s, UINT32_TO_PTR(__NR_poll + 1), INT_TO_PTR(-1)) >= 0);
-#elif defined __NR_ppoll
+                log_debug("has poll()");
+#endif
+#if defined __NR_ppoll && __NR_ppoll >= 0
                 assert_se(hashmap_put(s, UINT32_TO_PTR(__NR_ppoll + 1), INT_TO_PTR(-1)) >= 0);
+                log_debug("has ppoll()");
+#endif
+#if defined __NR_ppoll_time64 && __NR_ppoll_time64 >= 0
+                assert_se(hashmap_put(s, UINT32_TO_PTR(__NR_ppoll_time64 + 1), INT_TO_PTR(-1)) >= 0);
+                log_debug("has ppoll_time64()");
 #endif
 
-                assert_se(seccomp_load_syscall_filter_set_raw(SCMP_ACT_ALLOW, s, SCMP_ACT_ERRNO(EUNATCH), true) >= 0);
-
-                assert_se(access("/", F_OK) < 0);
-                assert_se(errno == EILSEQ);
+                if (hashmap_isempty(s)) {
+                        log_info("None of poll(), ppoll(), and ppoll_time64() defined, skipping several tests.");
+                        test_poll = false;
```

Comment:
I don't think this can happen. According to https://marcin.juszkiewicz.com.pl/download/tables/syscalls.html, every architecture has either some subset of those defined. And even for a new arch, it'd be really hard not to have those, since it's such an essential part of the Linux api. A simple `assert_se(!hashmap_is_empty(s))` should suffice.

Reply:
Yeah, I thought that's overkill.